Added available methods for has_one association to guide.

paulreece opened the pull request:
This adds the following methods to the definition of available methods in the guide: 
- `belongs_to` - `foreign_type`, `:required`, `:default`, `:strict_loading`, and `:ensuring_owner_was` .
- `has_one`  - `:required`, `:disable_joins`, `:strict_loading`,  `:query_constraints`, and `:ensuring_owner_was` .
- `has_many` - `:disable_joins`, `:ensuring_owner_was`, `:extend`, `:foreign_type`, `:query_constraints`, `:strict_loading`
- `has_and_belongs_to_many` - `:strict_loading`

### Motivation / Background

Addresses #49419.

### Detail


### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

skipkayhil commented:
I think we're also missing `query_constraints` here based on the API docs: https://edgeapi.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_one

These options also appear to be missing from the other associations as well, can we add them all in one go?

paulreece commented:
Done and done.

skipkayhil commented:
> These options also appear to be missing from the other associations as well, can we add them all in one go?

Can you do the other associations as well? I think they are also missing options

paulreece commented:
Sure 😎

paulreece committed:
This adds the following methods to the definition of available methods in the guide:

belongs_to - foreign_type, :required, :default, :strict_loading, and :ensuring_owner_was .
has_one - :required, :disable_joins, :strict_loading, :query_constraints, and :ensuring_owner_was .
has_many - :disable_joins, :ensuring_owner_was, :extend, :foreign_type, :query_constraints, :strict_loading
has_and_belongs_to_many - :strict_loading .
